doc/Overview/Mvux/Advanced/Pagination.md

```diff
@@ -0,0 +1,398 @@
+- If there's available space in the `ListView`, the next batch of 20 items will be requested from the Feed and loaded from the service thereafter.
+- When the user scroll down and hits the bottom of the `ListView`, the next 20 items will be requested.
+- This behavior will follow as the user scrolls down and chases the service for more data until all items have been loaded.
+
+Here's what the app renders like:
+
+![](../Assets/PaginationIncrementalLoading.gif)
+
+> [!TIP]  
+> You can use these members of the `ListView` to control data loading: [`DataFetchSize`](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.listviewbase.datafetchsize), [`IncrementalLoadingThreshold`](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.listviewbase.incrementalloadingthreshold), and [`IncrementalLoadingTrigger`](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.listviewbase.incrementalloadingthreshold).
+
+> [!NOTE]  
+> The source-code for the sample app demonstrated in this section can be found [here](https://github.com/unoplatform/Uno.Samples/tree/master/UI/MvuxHowTos/PaginationPeopleApp).
+
+## Offset pagination
+
+Offset pagination is controlled via a State that stores the current page index in the Model, and the List-Feed depending on it, using [the `Select` operator](xref:Overview.Mvux.Feeds#select-or-selectasync).  
+When the user requests a new page, the current page index state is updated, thereby updating the dependent collection List-Feed.
+
+Using the example started in [incremental loading above](#incremental-loading), we'll add another method to the service, which will disclose to the View how many items are there in total. Getting a count of items is more efficient than enumerating all entries. This is necessary to identify the total number of pages we have.
+
+### Service
+
+Here's the method added to the service (*PeopleService.cs*):
+
+```csharp
+public async ValueTask<int> GetPageCount(uint pageSize, CancellationToken ct) =>
+    (uint)Math.Ceiling(GetPeople().Length / (double)pageSize);
+```
+
+The signature of `GetPageCount` should also be added to the `IPeopleService` interface:
+
+```csharp
+ValueTask<int> GetPageCount(uint pageSize, CancellationToken ct);
+```
+
+### Model
+
+Let's expand the Model with the following:
+
+```csharp
+public partial record PeopleModel(IPeopleService PeopleService)
+{
+    public IListFeed<Person> PeopleAuto ...// will not be used in this example
+
+    const uint PageSize = 20;
+        
+    public IFeed<int> PageCount =>
+        Feed.Async(async (ct) => await PeopleService.GetPageCount(PageSize, ct));
+
+    public IState<uint> CurrentPage => State.Value(this, () => 1u);
+
+    public IListFeed<Person> PeopleManual =>
+        CurrentPage.SelectAsync(async (currentPage, ct) =>
+            // currentPage argument as index based
+            await PeopleService.GetPeopleAsync(pageSize: PageSize, pageIndex: currentPage - 1, ct))
+        .AsListFeed();
```

Comment:
```suggestion
    public IListFeed<Person> PeopleManual => CurrentPage
        // currentPage argument as index based
        .SelectAsync(async (currentPage, ct) => await PeopleService.GetPeopleAsync(pageSize: PageSize, pageIndex: currentPage - 1, ct))
        .AsListFeed();
```

Much more readable IMO. My rule is either no wrapping (if line is short), either add a new line for each operator.

Reply:
@dr1rrb

> Much more readable IMO. My rule is either no wrapping (if line is short), either add a new line for each operator.

Actually my rule when using functional / fluent **every** operator in new line regardless of length of line, and the class itself as well.

How about:

```suggestion
    public IListFeed<Person> PeopleManual => 
        CurrentPage
            // currentPage argument as index based
            .SelectAsync(async (currentPage, ct) => await PeopleService.GetPeopleAsync(pageSize: PageSize, pageIndex: currentPage - 1, ct))
            .AsListFeed();
```

BTW, In my personal code I'd do:

```c#
    public IListFeed<Person> PeopleManual => 
        CurrentPage
        // currentPage argument as index based
        .SelectAsync(async (currentPage, ct) =>
            await PeopleService
                  .GetPeopleAsync(
                      pageSize: PageSize, 
                      pageIndex: currentPage - 1, 
                      ct))
        .AsListFeed();
```
I know it's a bit extreme.

Reply:
> Much more readable IMO. 

I disagree. `CurrentPage` appearing on the right side of the lambda operator is pretty much invisible.

>  My rule is either no wrapping (if line is short), either add a new line for each operator.

It's also my rule, but I'm also placing the container in a new line. I'll update code accordingly.

Reply:
The main point here is to wrap before the `.SelectAsync` and keep the inner delegate on a single line as it's a method body. I would wrap a delegate only if multi-line, i.e. with curly brackets.

For the `CurrentPage` not being visible, I don't really agree, for me it makes clearer that the `PeopleManual` is derived from the `CurrentPage` (`PeopleManual => CurrentPage` being on a single line) on which we apply some rules (that are on lines below).

My rule for expression-bodied members is to keep them on a single line for property:
```csharp
public string A => _a.ToString();`
```
and wrap before the arrow for methods:
```csharp
public string GetA()
    => _a.ToString();
```

But all of those is only personal opinion (and I against strong rules for those!) Every dev as it's own way to think / work and as long as the alignment makes sense I'm fine with it :)